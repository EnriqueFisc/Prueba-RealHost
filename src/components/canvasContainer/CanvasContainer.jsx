import React from "react";
import { useSelectWall } from "../../hooks/useSelectWall";
import { ButtonsBar } from "../buttonsBar/ButtonsBar";

export const CanvasContainer = ({ children }) => {
  const { wall, selectWall } = useSelectWall();
  return (
    <>
      <div className="controls__container">
        <h1>Prueba t√©cnica RealHost</h1>
        <ButtonsBar selectWallFunc={selectWall} />
      </div>
      <section className="canvas__canvas-container">{children(wall)}</section>
    </>
  );
};
