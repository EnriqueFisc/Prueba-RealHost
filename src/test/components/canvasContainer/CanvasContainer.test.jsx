import { render } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import { CanvasContainer } from "../../../components/canvasContainer/CanvasContainer";

describe("Test in CanvasContainer", () => {
  test("should match with snapshot", () => {
    const { container } = render(<CanvasContainer>{() => {}}</CanvasContainer>);
    expect(container).toMatchSnapshot();
  });
});
