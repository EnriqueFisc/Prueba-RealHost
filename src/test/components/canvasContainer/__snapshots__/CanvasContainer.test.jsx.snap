// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test in CanvasContainer > should match with snapshot 1`] = `
<div>
  <div
    class="controls__container"
  >
    <h1>
      Prueba t√©cnica RealHost
    </h1>
    <div
      class="buttonsBar__container"
    >
      <button>
        Pared 1
      </button>
      <button>
        Pared 2
      </button>
      <button>
        Pared 3
      </button>
      <button>
        Pared 4
      </button>
    </div>
  </div>
  <section
    class="canvas__canvas-container"
  />
</div>
`;
